src/hooks/useActionCenterSubscriber/useTradeActionSubscriber.ts

```diff
@@ -0,0 +1,212 @@
+import { selectPendingSwapActionsFilteredByWallet } from '@/state/slices/selectors'
+import { selectSwapById, selectSwapByQuoteId } from '@/state/slices/swapSlice/selectors'
+import { swapSlice } from '@/state/slices/swapSlice/swapSlice'
+import {
+  selectFirstHopSellAccountId,
+  selectInputBuyAsset,
+  selectInputSellAsset,
+} from '@/state/slices/tradeInputSlice/selectors'
+import {
+  selectActiveQuote,
+  selectConfirmedTradeExecution,
+} from '@/state/slices/tradeQuoteSlice/selectors'
+import { TradeExecutionState } from '@/state/slices/tradeQuoteSlice/types'
+import { store, useAppDispatch, useAppSelector } from '@/state/store'
+
+export const useTradeActionSubscriber = () => {
+  const dispatch = useAppDispatch()
+  const translate = useTranslate()
+
+  const {
+    number: { toCrypto },
+  } = useLocaleFormatter()
+
+  const tradeExecution = useAppSelector(selectConfirmedTradeExecution)
+  const tradeSellAsset = useAppSelector(selectInputSellAsset)
+  const tradeBuyAsset = useAppSelector(selectInputBuyAsset)
+  const tradeQuote = useAppSelector(selectActiveQuote)
+  const previousTradeExecutionState = usePrevious(tradeExecution?.state)
+  const previousTradeQuoteId = usePrevious(tradeQuote?.id)
+  const currentHopIndex = useCurrentHopIndex()
+  const previousTxHash = usePrevious(tradeExecution?.firstHop.swap.sellTxHash)
+  const sellAccountId = useAppSelector(selectFirstHopSellAccountId)
+  const previousQuoteOrRate = usePrevious(tradeQuote?.quoteOrRate)
+  const pendingSwapActions = useAppSelector(selectPendingSwapActionsFilteredByWallet)
+  const toast = useToast()
+  const swapByIds = useAppSelector(selectSwapById)
+  const {
+    state: { isConnected },
+  } = useWallet()
+
+  // Create swap and action after user confirmed the intent
+  useEffect(() => {
+    if (!tradeExecution) return
+
+    const firstStep = tradeQuote?.steps[0]
+    const lastStep = tradeQuote?.steps[tradeQuote.steps.length - 1]
+
+    if (!firstStep || !lastStep) return
+    if (!sellAccountId) return
+
+    if (tradeQuote.quoteOrRate === 'quote' && previousQuoteOrRate === 'rate') {
+      const swap: Swap = {
+        id: uuidv4(),
+        createdAt: Date.now(),
+        updatedAt: Date.now(),
+        quoteId: tradeQuote.id as unknown as QuoteId,
+        sellTxHash: tradeExecution.firstHop.swap.sellTxHash,
+        sellAccountId,
+        swapperName: tradeQuote.swapperName,
+        sellAsset: firstStep.sellAsset,
+        buyAsset: lastStep.buyAsset,
+        sellAmountCryptoBaseUnit: tradeQuote.steps[0].sellAmountIncludingProtocolFeesCryptoBaseUnit,
+        buyAmountCryptoBaseUnit:
+          tradeQuote.steps[tradeQuote.steps.length - 1].buyAmountAfterFeesCryptoBaseUnit,
+        metadata: {
+          lifiRoute: isLifiTradeQuote(tradeQuote) ? tradeQuote.selectedLifiRoute : undefined,
+          chainflipSwapId: firstStep?.chainflipSpecific?.chainflipSwapId,
+          stepIndex: currentHopIndex,
+        },
+        status: SwapStatus.Pending,
```

Comment:
That looks incorrect, this isn't pending but idle.
We then use this in `selectPendingSwapActionsFilteredByWallet` which is consumed in `useTradeActionSubscriber` meaning we will poll for idle swaps

Reply:
It's been updated following our discussion, we now have a idle state!